packages/stateManager/index.ts

```diff
@@ -1,104 +1,478 @@
+    image: string;
+  };
+  tags: `0x${string}`[];
+  quantity: number;
 }
 
 interface Tag {
+  id: `0x${string}`;
   name: string;
-  items: Item[];
 }
 
 interface KeyCards {
   publicKey: `0x${string}`;
 }
+enum Status {
+  Failed = "FAILED",
+  Pending = "PENDING",
+  Complete = "COMPLETE",
+}
+interface Order {
+  id: `0x${string}`;
+  items: { [key: `0x${string}`]: number };
+  status: Status;
+  txHash?: string;
+  orderFinalized:
+    | {
+        orderHash: string;
+        currencyAddr: string;
+        totalInCrypto: string;
+        ttl: string;
+        payeeAddr: string;
+        shopSignature: string;
+        total: string;
+      }
+    | false;
+}
+interface IShopManifest {
+  name: string;
+  description: string;
+}
+interface UpdateShopManifest extends Partial<IShopManifest> {
+  setBaseCurrency?: {
+    tokenAddr: `0x${string}`;
+    chainId: number;
+  };
+  addAcceptedCurrency?: {
+    tokenAddr: `0x${string}`;
+    chainId: number;
+  };
+  addPayee?: {
+    addr: `0x${string}`;
+    callAsContract: boolean;
+    chainId: number;
+    name: string;
+  };
+  publishedTagId?: `0x${string}`;
+  profilePictureUrl?: string;
+}
 
-type ShopObjectTypes = Item | Tag | KeyCards;
+type ShopObjectTypes =
+  | Item
+  | Tag
+  | KeyCards
+  | Order
+  | (IShopManifest & UpdateShopManifest);
 
-// This is a an interface that is used to retrieve and store objects from a persistant layer
+// This is an interface that is used to retrieve and store objects from a persistant layer
 type Store<T extends ShopObjectTypes> = {
   put(key: string, value: T): Promise<void>;
   get(key: string): Promise<T>;
+  iterator: {
+    (options?: any): Iterator<string, ShopObjectTypes>;
+  };
 };
 
 abstract class PublicObjectManager<
   T extends ShopObjectTypes,
 > extends EventEmitter {
   constructor(
-    protected db: Store<T>,
+    protected store: Store<T>,
     protected client: RelayClient,
   ) {
     super();
   }
-  abstract _processEvent(event: schema.ShopEvents): Promise<boolean>;
-  abstract create(obj: T): Promise<[any, any]>;
-  abstract update(obj: T): Promise<[any, any]>;
-  get(id: string): Promise<T> {
-    return this.db.get(id);
+  abstract _processEvent(event: schema.ShopEvents): Promise<void>;
+  abstract get(key: string): Promise<T>;
+  abstract getIterator(): Store<T>["iterator"];
+}
+
+class ListingManager extends PublicObjectManager<Item> {
+  constructor(store: Store<Item>, client: RelayClient) {
+    super(store, client);
+  }
+  // Process all events for listings.
+  // Convert bytes to hex and save item object to listings store.
+  async _processEvent(event: schema.ShopEvents): Promise<void> {
+    if (event.createItem) {
+      const ci = event.createItem;
+      const id = bytesToHex(ci.eventId);
+      const item = {
+        id,
+        price: ci.price,
+        metadata: Uint8ArrayToParsedJSON(ci.metadata),
+        tags: [],
+        quantity: 0,
+      };
+      await this.store.put(id, item);
```

Comment:
```suggestion
      return this.store.put(id, item);
```

Reply:
Here we are always using `return` at the end of a control flow branch. This allows the reader of the code to know explicitly where the function stops executing.